### PR TITLE
chore(nix): remove nix subcommand, add appropriate errors

### PR DESCRIPTION
**Pull request by bketelsen**
if merged, this PR will remove the recently added `nix` subcommand and add a global `--nix` flag instead. subcommands that don't apply to nix will return an error if used.

### CODE REVIEW

File: cmd/remove.go [hunk 1]
```diff
@@ -32,7 +32,9 @@ func NewRemoveCommand() *cmdr.Command {
 }
 
 func remove(cmd *cobra.Command, args []string) error {
-
+	if cmd.Flag("nix").Changed {
+		return removePackage(cmd, args)
```
Comment (matbme) [inline cmd/remove.go:36]: I think these Nix functions should be moved to their respective commands to remove all `cmd/nix_[cmd].go` files.

**Review by kbdharun — APPROVED** [review]
Man page part and content LGTM, I haven't checked the code fully just glanced through it.

**Review by matbme — APPROVED** [review]
LGTM

File: man/apx.1 [hunk 2]
```diff
@@ -29,6 +29,8 @@ allowing you to install packages in a managed container\&.
       --zypper        Install packages from the OpenSUSE repository\&.
 .PP
       --xbps          Install packages from the Void (Linux) repository\&.
+.PP
+      --nix           Install packages from the Nix packages repository\&.
```
Comment (MonsterObserver) [inline man/apx.1:33]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: man/apx.1 [hunk 3]
```diff
@@ -155,6 +152,8 @@ Remove all unused packages automatically
...
```
Comment (MonsterObserver) [inline man/apx.1:156]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: man/apx.1 [hunk 4]
```diff
@@ -188,6 +187,8 @@ Clean the apx package manager cache
...
```
Comment (MonsterObserver) [inline man/apx.1:191]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: man/apx.1 [hunk 5]
```diff
@@ -219,6 +220,8 @@ Enter a shell in the managed container
...
```
Comment (MonsterObserver) [inline man/apx.1:224]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: man/apx.1 [hunk 6]
```diff
@@ -252,6 +255,8 @@ Export/Recreate a program's desktop entry from a managed container
...
```
Comment (MonsterObserver) [inline man/apx.1:259]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: man/apx.1 [hunk 7]
```diff
@@ -289,6 +294,8 @@ Initialize a managed container
...
```
Comment (MonsterObserver) [inline man/apx.1:298]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: man/apx.1 [hunk 8]
```diff
@@ -332,6 +339,8 @@ Install packages inside a managed container\&.
...
```
Comment (MonsterObserver) [inline man/apx.1:343]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: man/apx.1 [hunk 9]
```diff
@@ -371,6 +380,8 @@ List installed packages\&.
...
```
Comment (MonsterObserver) [inline man/apx.1:384]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: man/apx.1 [hunk 10]
```diff
@@ -402,6 +413,8 @@ Purge packages inside a managed container
...
```
Comment (MonsterObserver) [inline man/apx.1:417]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: man/apx.1 [hunk 11]
```diff
@@ -439,6 +452,8 @@ Remove packages inside a managed container\&.
...
```
Comment (MonsterObserver) [inline man/apx.1:456]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: man/apx.1 [hunk 12]
```diff
@@ -474,6 +489,8 @@ Run a program inside a managed container\&.
...
```
Comment (MonsterObserver) [inline man/apx.1:493]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: man/apx.1 [hunk 13]
```diff
@@ -509,6 +526,8 @@ Search for packages in a managed container\&.
...
```
Comment (MonsterObserver) [inline man/apx.1:530]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: man/apx.1 [hunk 14]
```diff
@@ -546,6 +565,8 @@ Show details about a package
...
```
Comment (MonsterObserver) [inline man/apx.1:569]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: man/apx.1 [hunk 15]
```diff
@@ -587,6 +608,8 @@ If --bin flag is provided, remove link to container binary\&.
...
```
Comment (MonsterObserver) [inline man/apx.1:612]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: man/apx.1 [hunk 16]
```diff
@@ -626,6 +649,8 @@ Update the list of available packages
...
```
Comment (MonsterObserver) [inline man/apx.1:653]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: man/apx.1 [hunk 17]
```diff
@@ -666,59 +691,8 @@ Update the list of available packages
...
-      --aur           Install packages from the AUR (Arch User Repository)\&.
-.PP
-      --dnf           Install packages from the Fedora's DNF (Dandified YUM) repository\&.
-.PP
-      --apk           Install packages from the Alpine repository\&.
-.PP
-      --zypper        Install packages from the OpenSUSE repository\&.
-.PP
-      --xbps          Install packages from the Void (Linux) repository\&.
+      --nix           Install packages from the Nix packages repository\&.
```
Comment (MonsterObserver) [inline man/apx.1:694]:
```suggestion
      --nix           Install packages from the Nixpkgs (Nix packages) repository\&.
```

File: locales/en.yml [hunk 18]
```diff
@@ -5,6 +5,7 @@ flags:
   apk: "Install packages from the Alpine repository."
   zypper: "Install packages from the OpenSUSE repository."
   xbps: "Install packages from the Void (Linux) repository."
+  nix: "Install packages from the Nix packages repository."
```
Comment (MonsterObserver) [inline locales/en.yml:8]:
```suggestion
  nix: "Install packages from the Nixpkgs (Nix packages) repository."
```

**Review by MonsterObserver — CHANGES_REQUESTED** [review]
Add Nixpkgs

**Review by MonsterObserver — APPROVED** [review]
LGTM

**Review by mirkobrombin — APPROVED** [review]

**Pull request merged**